respond to pr feedback on reindex_sql_forms_in_domain

dannyroberts opened the pull request:
from https://github.com/dimagi/commcare-hq/pull/13097

dannyroberts committed:
respond to pr feedback on reindex_sql_forms_in_domain

https://github.com/dimagi/commcare-hq/pull/13097

dannyroberts commented:
@snopoke

dannyroberts commented:
cc: @nickpell @sravfeyn

snopoke commented on corehq/apps/hqcase/management/commands/reindex_sql_forms_in_domain.py:
```diff
@@ -1,17 +1,21 @@
+        for doc_ids in chunked(all_doc_ids, 100):
+            print 'Reindexing doc_ids: {}'.format(','.join(doc_ids))
+            reindexer.doc_processor.process_bulk_docs([
+                transform_xform_for_elasticsearch(form.to_json(include_attachments=True))
```
Since you're using the reindexer it will handle the transformation for you so you don't need `transform_xform_for_elasticsearch`

dannyroberts replied:
Oh wow you're totally right. I was confusing the `doc_provider` (which doesn't know about the transform) and the `doc_processor` (which knows about both the doc provider and the transform).

This was wrong in my previous PR too. Thanks for catching, I'll change it

dannyroberts replied:
I also decided to use `reindexer.reindex_accessor.doc_to_json` instead of hardcoding `form.to_json(...)`

dannyroberts committed:
remove double transform in reindex_sql_forms_in_domain

also use reindexer.reindex_accessor.doc_to_json instead of hardcoding form.to_json

nickpell commented:
looks good to me
